### PR TITLE
Part of #17670 for file signature-request-data.js

### PR DESCRIPTION
**Pull request by PrgrmrHarshShukla**
## Explanation
This PR is a part of the issue #17670 Replace deprecated Typography with Text component for the file 
**ui\components\app\signature-request\signature-request-data\signature-request-data.js**

### Before

Screenshot 2023-07-24 at 12 30 00 PM

### After

Screenshot 2023-07-24 at 12 29 46 PM

https://github.com/MetaMask/metamask-extension/assets/8112138/7f022364-9937-4708-9bb5-5174fcd59ef9

## Manual Testing Steps
- Pull this branch
- Run storybook `yarn storybook`
- Check `SignatureRequestData` story against https://metamask.github.io/metamask-storybook/?path=/story/components-app-signaturerequest-signaturerequestdata--default-story
OR
- use the test dapp page https://metamask.github.io/test-dapp/
- Scroll down to Sign Typed Data V4
- Click SIGN
- See `SignatureRequestData`

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by PrgrmrHarshShukla** [comment]
Hey @georgewrmarshall ,
even though I made only the supposed changes, there is difference in the spacing between the components.
Please look into this.

**Review by georgewrmarshall — COMMENTED** [review]
This LGTM. I'm not sure why e2e tests are failing at first glance the files seem unrelated

**Comment by PrgrmrHarshShukla** [comment]
Hey @georgewrmarshall 
There are reduced spacings between different lines after the changes in the PR.
Is that OK?

**Comment by PrgrmrHarshShukla** [comment]
Hey @georgewrmarshall , 
We are struggling with this,
```
TimeoutError: Waiting for element to be located By(css selector, [data-testid=&#x22;signature-request-scroll-button&#x22;])
Wait timed out after 10149m
```


Probably we can consider increasing the timeout value in the test script.
I also felt that the elements were taking a long time to load initially and update after changes.

**Comment by georgewrmarshall** [comment]
Hey @PrgrmrHarshShukla, try rebasing from `develop` a PR just got merged that should fix some timeouts. I can't see immediately how the e2e tests are failing here it seems like it's failing on an unrelated component to these file changes

**Comment by PrgrmrHarshShukla** [comment]
Hey @georgewrmarshall , 

My forked is not behind MetaMask:develop
I use GitHub Desktop, it also shows nothing to pull form remote origin.

Screenshot (171)

**Comment by PrgrmrHarshShukla** [comment]
Hey @georgewrmarshall ,
I think I messed this up.
Please help.

File: ui/components/app/signature-request/signature-request-data/signature-request-data.js [hunk 1]
```diff
@@ -12,14 +12,14 @@ import {
   toChecksumHexAddress,
 } from '../../../../../shared/modules/hexstring-utils';
 import Box from '../../../ui/box';
```
Comment (NidhiKJha) [inline ui/components/app/signature-request/signature-request-data/signature-request-data.js:14]: Can you update this file to use the `Box` component from the component-library?

**Review by NidhiKJha — COMMENTED** [review]
Looks good. Left one suggestion

File: ui/components/app/signature-request/signature-request-data/signature-request-data.js (hunk 1 above)
Comment (PrgrmrHarshShukla) [reply ui/components/app/signature-request/signature-request-data/signature-request-data.js:14]: 👍
Will do.
Thanks for pointing out.

File: ui/components/app/signature-request/__snapshots__/signature-request.component.test.js.snap [hunk 2]
```diff
@@ -249,14 +249,14 @@ exports[`Signature Request Component render should match snapshot when we are us
             class="box signature-request-data__node box--padding-left-2 box--display-flex box--flex-direction-row"
           >
             <span
-              class="box box--margin-top-1 box--margin-bottom-1 box--margin-left-4 box--flex-direction-row typography typography--p typography--weight-normal typography--style-normal typography--color-text-default"
+              class="box mm-text mm-text--body-md mm-text--font-weight-normal box--margin-left-4 box--flex-direction-row box--color-text-default"
```
Comment (georgewrmarshall) [inline ui/components/app/signature-request/__snapshots__/signature-request.component.test.js.snap:252]: You may need to update snapshots I was expecting this to use the `mm-box` className?

**Review by georgewrmarshall — COMMENTED** [review]
Hey @PrgrmrHarshShukla, looks like you need to update snapshots and possibly look into e2e test failures.

File: ui/components/app/signature-request/__snapshots__/signature-request.component.test.js.snap (hunk 2 above)
Comment (PrgrmrHarshShukla) [reply ui/components/app/signature-request/__snapshots__/signature-request.component.test.js.snap:252]: Hey @georgewrmarshall ,
Updated the snapshots.👍
Please look into this.

**Comment by PrgrmrHarshShukla** [comment]
Hey @georgewrmarshall ,
I found this in the detailed report of the failing e2e test for chrome,
`Starting ChromeDriver 114.0.5735.90 (386bc09e8f4f2e025eddae123f36f6263096ae49-refs/branch-heads/5735@{#1052}) on port 44793`
and then,
`[1689826955.560][SEVERE]: bind() failed: Cannot assign requested address (99)`

The same on several ports.

I tried to search why that is happening I found these possible reasons,
1. The IP address specified for binding is not valid or does not exist on the machine.
2. The IP address specified is not associated with any network interface on the machine.
3. The port number is already in use by another process, preventing the socket from binding to it.

Cannot figure it out completely, need help.
Please look into this, may be we can find a solution.

**Comment by PrgrmrHarshShukla** [comment]
And, for the failing e2e test for firefox, 
I found this,
```
<testcase classname="Sign Typed Data Signature Request" name="can initiate and confirm a Signature Request of Sign Typed Data V3" time="24.187"><failure>Waiting for element to be located By(css selector, [data-testid=&#x22;signature-request-scroll-button&#x22;])
Wait timed out after 10070ms
TimeoutError: Waiting for element to be located By(css selector, [data-testid=&#x22;signature-request-scroll-button&#x22;])
Wait timed out after 10070ms
```

We encountered a similar timeout error earlier in a different PR.
What am I supposed to do?
Please look into this too.

**Comment by PrgrmrHarshShukla** [comment]
Hey @georgewrmarshall ,
I see that there are some conflicts here.
But I am not able to find them.
Also,
the `Resolve conflicts` button is inactive for me.

Please help.🙏


Screenshot_20230722_121831_Chrome

**Review by georgewrmarshall — APPROVED** [review]
Hey @PrgrmrHarshShukla, this looks great! I updated some of the semantic HTML here and rebased from develop. I've also updated the before/after screenshots in the description. LGTM Thanks for your contribution

**Review by garrettbear — APPROVED** [review]

**Pull request merged**